dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -764,7 +764,8 @@ def get_first_selected_workshop
       # Map back to actual workshops by reconstructing the friendly_date_range
       workshops = Pd::Workshop.where(id: workshop_ids)
       hash[:able_to_attend_multiple].each do |response|
-        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range)}
+        location = response.scan(/in (.+) hosted/).first.try(:first) || ''
+        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range) && w.location_address == location}
```

Comment:
This should be a 2-pass, using location as a fallback when there are multiple date matches. As we discussed yesterday, many workshop locations were TBD when the applications were submitted, with real locations filled in later. The dates _should_ be more reliable, and haven't changed AFAIK.

Reply:
That's fair - I'll change it